File: 01/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	inputFile, _ := os.Open("input.txt")
	defer inputFile.Close()

	lines := bufio.NewScanner(inputFile)

	calorieCounts := []int{}
	currentElfCount := 0
	for lines.Scan() {
		line := lines.Text()

		if line == "" {
			calorieCounts = append(calorieCounts, currentElfCount)
			currentElfCount = 0
			continue
		}

		calories, _ := strconv.Atoi(line)
		currentElfCount += calories
	}

	calorieCounts = append(calorieCounts, currentElfCount)

	first := 0
	second := 0
	third := 0
	for _, calorieCount := range calorieCounts {
		if calorieCount > third {
			third = calorieCount
		}

		if calorieCount > second {
			second, third = calorieCount, second
		}

		if calorieCount > first {
			first, second = calorieCount, first
		}
	}

	fmt.Println(first)
	fmt.Println(second)
	fmt.Println(third)

	fmt.Println(first + second + third)
}

File: 06/part1.py
def firstMarkerPosFor(line):
    for idx, char in enumerate(line):
        if startOfPacket(idx, line):
            return idx + 1

    return None


def startOfPacket(idx, line):
    if idx < 4:
        return False

    checkedChars = set()

    for i in range(4):
        if line[idx-i] in checkedChars:
            return False

        checkedChars.add(line[idx-i])

    return True


with open('input.txt') as file:
    for line in file:
        print(firstMarkerPosFor(line))
